lib/rules/no-useless-escape.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @fileoverview Look for useless escapes in strings and regexes
+ * @author Onur Temizkan
+ * @copyright 2016 Onur Temizkan. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var VALID_STRING_ESCAPES = {
+    "\\": true,
+    "n": true,
+    "r": true,
+    "v": true,
+    "t": true,
+    "b": true,
+    "f": true,
+    "u": true,
+    "x": true
+};
+
+var VALID_REGEX_ESCAPES = {
+    "\\": true,
+    "^": true,
+    "$": true,
+    "*": true,
+    "+": true,
+    "?": true,
+    ".": true,
+    "{": true,
+    "[": true,
+    "|": true,
+    "(": true,
+    "b": true,
+    "B": true,
```

Comment:
It appears there's not actually a `B` class escape in JavaScript regexen. See https://tc39.github.io/ecma262/#prod-ClassEscape